packages/peregrine/lib/hooks/useDropdown.js

```diff
@@ -15,7 +15,9 @@ export const useDropdown = () => {
 
     // collapse on mousedown outside of this element
     const maybeCollapse = useCallback(({ target }) => {
-        if (!elementRef.current.contains(target)) {
+        const isValidElementRef = elementRef && elementRef.current;
+
+        if (isValidElementRef && !elementRef.current.contains(target)) {
```

Comment:
`useRef` always returns an object shape `{current: <value>}`, you should just need a truthy check of current since your initialValue is null.

```suggestion
        if (elementRef.current && !elementRef.current.contains(target)) {
```

If we wanted to be extra careful, you could even do a type check to ensure contains is in the prototype; `elementRef.current instanceof Node` (HTMLElement might be better?).

Reply:
I think type check can be an overkill since it is only created and used by react/react components, the type should/will not change. Regarding the `current` check, I am doing that as part of `isValidElementRef`. Is there something that I am missing?

Reply:
> `useRef` always returns an object shape `{current: <value>}`

Tommy is saying that you don't need the `elementRef` part of the condition because it's unnecessary. And if you don't need the `elementRef` truthy check, assigning `elementRef.current` to a variable is just extra. That's his suggested change.

He adds that you can protect the condition against running on invalid ref assignments by verifying that the ref's current value is of the type expected.